packages/core-data/src/resolvers.js

```diff
@@ -450,6 +450,15 @@ export const canUser =
 					.join( '/' );
 
 				dispatch.receiveUserPermission( key, permissions[ action ] );
+
+				// Mark related action resolutions as finished.
+				if ( action !== requestedAction ) {
+					dispatch.finishResolution( 'canUser', [
```

Comment:
So you're saying, in some scenarios, this could invalidate the existing check at the top?

https://github.com/WordPress/gutenberg/blob/da558e272d60a7de6f823032763231be56cb6410/packages/core-data/src/resolvers.js#L397-L410

> A better approach might be to separate this resolver into a sub selector/resolver no?

Do you have an example?

Reply:
No, actually I missed that check and you're right, we seem covered already. 

That said, I think we can avoid all of that (both marking as resolved and the check to avoid starting the resolution) by relying on a shared sub resolver/selector.

Reply:
> That said, I think we can avoid all of that (both marking as resolved and the check to avoid starting the resolution) by relying on a shared sub resolver/selector.

Making a note and happy to follow up on this as "polish step". While I think I have an idea of what you mean by shared sub-resolver/selector. Do you mind sharing an example so we're on the same page 😅

Reply:
Something like:

```
export const canUser =
	( requestedAction, resource, id ) => async ( { resolveSelect } ) =>  {
		await resolveSelect.getResourcePermissions( resource, id );
	}
		
```